text/3724-theupdateframework.md

```diff
@@ -0,0 +1,341 @@
+
+## Summary & Motivations
+
+We propose the creation of two distinct TUF repositories for signing of Rust Project content and crates, respectively. Two main motivations exist for separating these concerns: The cadence of content published within each, and the trust of each. Rustup and Rust releases (both nightly and stable) are conducted under a controlled and predictable manner which is managed by the Project. However, crates are published by the community, and as such we see a larger and much more varied volume of content which may exist within this repository. We have additionally modeled signing the root of one repository by the other - this implicitly grants us a chain of trust from the "Project" (tuf-root) to the separate crates.io repository (tuf-crates). The sections below go into more detail on each repository and its configuration.
+
+### `rust-lang/tuf-root`
+
+tuf-root shall be used for signing Rust releases, rustup releases, and an independent and updated version of the tuf-crates root role metadata file (root.json).
+
+#### Roles
+
+##### Root Role
+
+The root role of the tuf-root shall be a TUF role consisting of 9 members with a 5 member threshold for signing (5-of-9); please reference the Root Quorum Model section below for details on how this role should be managed and its members selected. The sole purpose of this role shall be delegating authority to the other roles within the tuf-root repository (when members of these roles change). Finally, this role shall also be used for signing the tuf-crates root.json - thus protecting the chain of trust between tuf-root and tuf-crates.
+
+##### Targets Role
+
+The repository shall have a top-level Targets role, as specified by TUF, which then delegates authority to the following delegate target roles:
+
+##### Release (Stable/Beta) Role
+
+The Release role shall have the authority to only sign stable rust releases. We propose this role also consist of a quorum model, consisting of all members of the release team. This role should have a 3 member threshold, and always consist of all members of the release team. At the time a new stable release is being compiled and shipped, a signing quorum must be conducted for this release.
+
+###### Nightly Role
+
+Nightly releases shall be conducted by a single signing key, trusted for only signing nightly releases of the Rust Compiler. This will allow for nightly releases to remain automated and not require the active participation of any rust members.
+
+###### Rustup Role
+
+This shall be a quorum based role, consisting of all members of the Rustup & Infrastructure team members. We recommend having at least a 3-member threshold. We have decided to have this roles quorum be broader to allow for emergency updates and releases of Rustup; we may want to increase the threshold when these teams have more members.
+
+###### Snapshots & Timestamps Role
+
+These roles shall be a single-member role with a key utilized for automation.
```

Comment:
what are these roles used for? in other words: what does "snapshots & timestamps" mean in this context?

Reply:
They are technical terms of art from the TUF specification. The `snapshot` is a single representation of every file in the index with the current hash for that file. This is used to ensure that the mirror/MITM is not holding back some individual package while keeping the others up to date. The `timestamp` is a very small file that points to the latest version of the other files, whose signature is very short-lived. This prevents the mirror/MITM from staying stuck on an old snapshot.